docs/flows/flows.puml

```diff
@@ -272,20 +271,136 @@ bitdaoadmin <-- insurancecontract: execute proposal
 
 newpage
 
-autonumber "Upgrades:[00]"
+autonumber "UpgradeGovernance:[00]"
 
-== Upgrades ==
+== Upgrade Governance ==
 
-bitdaoadmin -> governanceproxy: deploy proxy and set initial implementation
-bitdaoadmin -> treasuryproxy: deploy proxy and set initial implementation
-bitdaoadmin -> executorproxy: deploy proxy and set initial implementation
+bitdaoadmin -> executor: deploy timelock executor
+bitdaoadmin -> bitdaogovernance: deploy initial governance contract logic
+bitdaoadmin -> proxyadmin: deploy admin for governance proxy
+bitdaoadmin -> proxy: deploy governance proxy and set logic and admin
+bitdaoadmin -> proxyadmin: set executor as admin
```

Comment:
issue: each call should probably return values to the bitdaoadmin.

suggestion: consider a helper contract to avoid the bitdaoadmin from dealing with addresses or performing these actions step by step

Reply:
Those are separate contracts deployments with upgrades happening rarely. I'd say its not helper contract but rather deployment/upgrade script, which makes all those steps with single execution.